Framework/Geometry/test/MeshObjectTest.h

```diff
@@ -0,0 +1,1089 @@
+    triangles.push_back(5);
+    triangles.push_back(4);
+    // z max
+    triangles.push_back(6);
+    triangles.push_back(7);
+    triangles.push_back(11);
+    triangles.push_back(11);
+    triangles.push_back(9);
+    triangles.push_back(10);
+    triangles.push_back(9);
+    triangles.push_back(7);
+    triangles.push_back(8);
+    // y min
+    triangles.push_back(0);
+    triangles.push_back(1);
+    triangles.push_back(6);
+    triangles.push_back(6);
+    triangles.push_back(1);
+    triangles.push_back(7);
+    // x max
+    triangles.push_back(1);
+    triangles.push_back(2);
+    triangles.push_back(7);
+    triangles.push_back(7);
+    triangles.push_back(2);
+    triangles.push_back(8);
+    // y mid
+    triangles.push_back(2);
+    triangles.push_back(3);
+    triangles.push_back(8);
+    triangles.push_back(8);
+    triangles.push_back(3);
+    triangles.push_back(9);
+    // x mid
+    triangles.push_back(3);
+    triangles.push_back(4);
+    triangles.push_back(9);
+    triangles.push_back(9);
+    triangles.push_back(4);
+    triangles.push_back(10);
+    // y max
+    triangles.push_back(4);
+    triangles.push_back(5);
+    triangles.push_back(10);
+    triangles.push_back(10);
+    triangles.push_back(5);
+    triangles.push_back(11);
+    // x min
+    triangles.push_back(5);
+    triangles.push_back(0);
+    triangles.push_back(11);
+    triangles.push_back(11);
+    triangles.push_back(0);
+    triangles.push_back(6);
+
+    boost::shared_ptr<MeshObject> retVal =
+        boost::shared_ptr<MeshObject>(new MeshObject);
+    retVal->initialize(triangles, vertices);
+    return retVal;
+  }
+};
+
+// -----------------------------------------------------------------------------
+// Performance tests
+// -----------------------------------------------------------------------------
+class MeshObjectTestPerformance : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static MeshObjectTestPerformance *createSuite() {
+    return new MeshObjectTestPerformance();
+  }
+  static void destroySuite(MeshObjectTestPerformance *suite) { delete suite; }
+
+  MeshObjectTestPerformance()
+      : rng(200000), octahedron(createOctahedron()), lShape(createLShape()) {}
+
+  void test_generatePointInside_Convex_Solid() {
+    const size_t maxAttempts(500);
+    V3D dummy;
+    for (size_t i = 0; i < npoints; ++i) {
+      dummy = octahedron->generatePointInObject(rng, maxAttempts);
+    }
+  }
+
+  void test_Point_Inside_NonConvex_Solid() {
+    const size_t maxAttempts(500);
+    V3D dummy;
+    for (size_t i = 0; i < npoints; ++i) {
+      dummy = lShape->generatePointInObject(rng, maxAttempts);
+    }
+  }
+
+  boost::shared_ptr<MeshObject> createOctahedron() {
+    /**
+    * Create octahedron with vertices on the axes at -1 & +1.
+    */
+    boost::shared_ptr<MeshObject> retVal =
```

Comment:
1. `boost::make_shared`,  not `new MeshObject`
1. but should be a `unique_ptr` anyway as no sharing needed.
1. Description does not match comments for this. Does this really return an Octohedron?
1. Either make this a `const` method or define it as a non-member function

Reply:
If you think this does not return an octahedron, then exactly what do you think it returns. Note that opposite vertices have indices differing by 3. I'll mention that in the comments.

Reply:
This is not creating an Octahedron. This is creating a NULL/Invalid/empty shape of some sorts. This is the implementation of `createOctahdron` in your `MeshObjectPerformanceTest`. You also have one in your `MeshObjectTest` line 885, which is doing the right thing. Maybe it would be best to define a free function for `createOctohedron` that would be visible to tests in both test classes.